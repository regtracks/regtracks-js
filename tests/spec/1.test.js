const Track = require('../../index');

describe('tests section 1 of the specification', () => {
  test('section 1.1: unicode', () => {
    let schema = String.raw`
    @foobar
    ||
    any Ϣ, ח, ඐ, a, b, c, (䔘䬨)
    or (䰅䰅䰅)
    ||
    `;

    const track = new Track(schema);

    expect(track.parser.tree.tree).toMatchSnapshot();
  });
});
