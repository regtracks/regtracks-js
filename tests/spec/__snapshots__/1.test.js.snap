// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests section 1 of the specification section 1.1: unicode 1`] = `
Object {
  "foobar": Object {
    "name": "foobar",
    "root": Object {
      "children": Array [
        Object {
          "collectionIdent": undefined,
          "forks": false,
          "inclusive": true,
          "matches": Array [
            Object {
              "matches": "Ϣ",
              "type": "literal_criterion",
            },
            Object {
              "matches": "ח",
              "type": "literal_criterion",
            },
            Object {
              "matches": "ඐ",
              "type": "literal_criterion",
            },
            Object {
              "matches": "a",
              "type": "literal_criterion",
            },
            Object {
              "matches": "b",
              "type": "literal_criterion",
            },
            Object {
              "matches": "c",
              "type": "literal_criterion",
            },
            Object {
              "matches": "䔘䬨",
              "type": "literal_criterion",
            },
          ],
          "notMatch": false,
          "optional": false,
          "repeat": undefined,
          "type": "matching_rule",
        },
        Object {
          "collectionIdent": undefined,
          "forks": true,
          "inclusive": true,
          "matches": Array [
            Object {
              "matches": "䰅䰅䰅",
              "type": "literal_criterion",
            },
          ],
          "notMatch": false,
          "optional": false,
          "repeat": undefined,
          "type": "matching_rule",
        },
      ],
      "collectionIdent": undefined,
      "forks": false,
      "notMatch": false,
      "optional": false,
      "parent": undefined,
      "repeat": undefined,
      "type": "block",
    },
    "type": "pattern",
  },
}
`;
